### PR TITLE
#163447376 Add Soft Delete to the Application

### PR DESCRIPTION
**Pull request by michael-basweti**
#### What does this PR do?
* Added soft delete to all delete operations.
#### Description of Task to be completed?
* Added a `status` field to all models that have a delete functionality.
* Altered queries, Update mutations and delete mutations in schemas to access data that is only active in the database
#### How should this be manually tested?
* Clone the branch and follow the following steps to [setup](https://github.com/andela/mrm_api).
* git checkout to `ch-implement-soft-delete-163447376`
* Create different resources i.e location, rooms and offices, A status field will be added in each and every relation in the database. When you delete anything, its status will be changed from active to archived. 
#### Any background context you want to provide?
* Not applicable
#### What are the relevant pivotal tracker stories?
[#163447376](https://www.pivotaltracker.com/n/projects/2154921/stories/163447376)

### CODE REVIEW

File: api/office/schema.py [hunk 1]
```diff
@@ -113,17 +119,18 @@ def resolve_all_offices(self, info, **kwargs):
...
         if not check_office:
             raise GraphQLError("Office Not found")
 
         if name == "Epic tower":
-            exact_query = lagos_office_join_location(query)
+            exact_query = lagos_office_join_location(resultActive)
             result = exact_query.filter(OfficeModel.name == name)
             return result.all()
 
         else:
```
Comment (kampkelly) [inline api/office/schema.py]: You can remove this unnecessary blank line here.

File: api/floor/schema.py [hunk 2]
```diff
@@ -78,19 +80,22 @@ def mutate(self, info, floor_id, **kwargs):
...
     class Arguments:
+        status = graphene.String()
         floor_id = graphene.Int(required=True)
     floor = graphene.Field(Floor)
 
     @Auth.user_roles('Admin')
     def mutate(self, info, floor_id, **kwargs):
         query_floor = Floor.get_query(info)
-        exact_floor = query_floor.filter(
+        resultActive = query_floor.filter(FloorModel.status == "active")
```
Comment (omobosteven) [inline api/floor/schema.py]: You did not follow the naming convention for this variable and also the name could be more descriptive.

Comment (michael-basweti) [reply api/floor/schema.py]: @omobosteven , I have implemented the changes

File: api/office/schema.py (hunk 1 above)
Comment (michael-basweti) [reply api/office/schema.py]: @kampkelly , which line specifically?

Comment (kampkelly) [reply api/office/schema.py]: Line 131 @michael-basweti

File: api/location/schema.py [hunk 3]
```diff
@@ -72,8 +74,9 @@ def mutate(self, info, location_id, **kwargs):
             validate_url(**kwargs)
         if "abbreviation" in kwargs:  # noqa
             validate_empty_fields(**kwargs)
-        result = location.filter(LocationModel.name == kwargs.get('name'))
-        if result.count() > 0:
+        active_locations = result.filter(
+            LocationModel.name == kwargs.get('name'))
+        if active_locations.count() > 0:
             pass
```
Comment (kampkelly) [inline api/location/schema.py]: @michael-basweti you can just do `if active_locations.count()` instead of `if active_locations.count() > 0`

Comment (michael-basweti) [reply api/location/schema.py]: @kampkelly We need this for readability and also we should have an active location for operation to go through

File: api/office/schema.py (hunk 1 above)
Comment (michael-basweti) [reply api/office/schema.py]: @kampkelly there is no blank line at 131

File: api/location/schema.py [hunk 4]
```diff
@@ -72,8 +75,9 @@ def mutate(self, info, location_id, **kwargs):
             validate_url(**kwargs)
         if "abbreviation" in kwargs:  # noqa
             validate_empty_fields(**kwargs)
-        result = location.filter(LocationModel.name == kwargs.get('name'))
-        if result.count() > 0:
+        active_locations = result.filter(
+            LocationModel.name == kwargs.get('name'))
+        if active_locations.count() > 0:
```
Comment (Bonifase) [inline api/location/schema.py]: Why is counting necessary at this point? Could it be possible to check active locations without counting?

Comment (michael-basweti) [reply api/location/schema.py]: @Bonifase ,I have implemented the changes.

File: alembic/versions/62b9fc80f1fe_add_soft_delete.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+
+# revision identifiers, used by Alembic.
+revision = '62b9fc80f1fe'
+down_revision = 'c53edf85cda4'
```
Comment (RonKbS) [inline alembic/versions/62b9fc80f1fe_add_soft_delete.py]: This down revision references a the wrong file. There is a new file on develop with the number `c8a866f85fd7` that it should show instead.
@RachaelNantale has tried it out on her p.c as well and got the same error

**Comment by michael-basweti** [comment]
> Please fix the alembic issues.

I have done that.

**Comment by michael-basweti** [comment]
> @michael-basweti let's go with `state` for the column name. `status` is more of an informatory flag while `state` insinuates an action to be performed depending on the state value.
> Also, please give your enum variable a more appropriate name.

@joshuaocero I have implemented the changes

**Pull request merged**